### PR TITLE
Create a common library for route-agent vxlan driver and vxlan cable driver

### LINKED ISSUE #1393: Create a common library for route-agent vxlan driver and vxlan cable driver
**What would you like to be added**:
Create a common library for routeagent vxlan driver and vxlan cable driver.
**Why is this needed**:

### PR DESCRIPTION
**Pull request by tpantelis**
See individual commits for details.

Fixes https://github.com/submariner-io/submariner/issues/1393

### CODE REVIEW

**Review by dfarrell07 — APPROVED** [review]

File: pkg/vxlan/vxlan.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+			LinkIndex: i.link.Index,
+			Src:       srcIP,
+			Dst:       &destCIDRs[j],
+			Gw:        gwIP,
+			Type:      netlink.NDA_DST,
+			Flags:     netlink.NTF_SELF,
+			Priority:  100,
+			Table:     tableID,
+		}
+		err := i.netLink.RouteAdd(route)
```
Comment (yboaron) [inline pkg/vxlan/vxlan.go]: Can't we use netlinkAPI.RouteAddOrReplace here ?

File: pkg/vxlan/vxlan.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+			Gw:        nil,
+			Type:      netlink.NDA_DST,
+			Flags:     netlink.NTF_SELF,
+			Priority:  100,
+			Table:     tableID,
+		}
+
+		err := i.netLink.RouteDel(route)
+		if err != nil {
+			return errors.Wrapf(err, "unable to add the route entry %v", route)
```
Comment (yboaron) [inline pkg/vxlan/vxlan.go]: nit , should be 'delete'

File: pkg/vxlan/vxlan.go [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+			LinkIndex: i.link.Index,
+			Dst:       &destCIDRs[j],
+			Gw:        nil,
+			Type:      netlink.NDA_DST,
+			Flags:     netlink.NTF_SELF,
+			Priority:  100,
+			Table:     tableID,
+		}
+
+		err := i.netLink.RouteDel(route)
```
Comment (yboaron) [inline pkg/vxlan/vxlan.go:259]: Don't we want to return nil if route doesn't exist ? ,  like 
https://github.com/submariner-io/submariner/blob/devel/pkg/routeagent_driver/handlers/ovn/gateway_dataplane.go#L48

File: pkg/vxlan/vxlan.go (hunk 1 above)
Comment (tpantelis) [reply pkg/vxlan/vxlan.go]: yes

File: pkg/vxlan/vxlan.go (hunk 3 above)
Comment (tpantelis) [reply pkg/vxlan/vxlan.go:259]: yeah that should be fine here.

**Review by yboaron — APPROVED** [review]

**Pull request merged**